### ISSUE TITLE
Shell completions

### ISSUE THREAD

**Issue by ulken** [Evidlo/passhole#44]
This is probably quite cumbersome, but would be nice with shell (bash/zsh/fish) completions populated with matching entries from the database 🙂.

Thoughts? Do you have any experience with generating necessary data in Python? Feasible or not?

**Comment by ulken** [comment]
I'm gonna give this a go, using `argcomplete`. I like `shtab`'s philosophy better, but doesn't seem to support dynamic completions which would be required here.

No pressure on merging if you disapprove. If it works reasonably well, I'll just maintain my own fork.

**Comment by Evidlo** [comment]
I thought about this a long time ago (https://github.com/Evidlo/passhole/issues/2), but passhole didn't have database caching at the time so it would have required entering your password for each completion.

Take a shot at it.  I imagine you want entry/group completion for mv/ls/show/edit/rm?

**Comment by ulken** [comment]
Makes sense. Even now, we still need to login once before completions can work. I'm thinking about adding a `login` command. I don't think we can prompt for password when tab-completing*. And it feels weird to run an arbitrary command just to cache the db. How do you feel about that?

> Take a shot at it. I imagine you want entry/group completion for mv/ls/show/edit/rm?

Yeah, basically all commands taking a `path` as argument. 

* Although, we can `warn` about login being required for completions to work.
